### PR TITLE
Neptune add support for 3.7

### PR DESCRIPTION
**Pull request by cloutierMat**
# Motivation

Github issue localstack/localstack#10823 request for gremlin transaction requires support for gremlin server 3.7.2.

# Changes

- Added a `UUIDAnyIDManager` build for version 3.7.x as the `Idmanager` has moved from `TinkerGraph.IdManager` to `AbstractTinkerGraph.IdManager`
- Upgraded gradle build for the IdManager to support multiple builds
- Added label delimiter patch for version 3.7.2

## Also

- Removed support for deprecated version 3.4.3. Users still using an older localstack version that allows installing 3.4.3 will not suffer from this as the package installer points at the commit hashes.

### CODE REVIEW

File: tinkerpop-id-manager/3.7.jar
Comment (whummer) [inline tinkerpop-id-manager/3.7.jar:1]: Looks like this is an empty file - could be removed?

File: neptune-tinkerpop/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 *.class
 target/
+*.kryo
```
Comment (whummer) [inline neptune-tinkerpop/.gitignore:3]: Out of curiosity - did you end up seeing some `*.kryo` files locally? 

Reason I'm asking - we did use kryo to support [persistence for the old StepFunctions provider](https://github.com/localstack/localstack-artifacts/blob/master/stepfunctions-local-patch/cloud/localstack/PersistenceContext.java#L35-L61), prior to releasing our StepFunctions v2 rewrite in pure Python. Was just curious if Gremlin also uses kryo - and seems like it, according to [the docs here](https://tinkerpop.apache.org/docs/current/reference/#gryo).

**Review by whummer — APPROVED** [review]
Great set of changes, kudos for adding this @cloutierMat ! 🚀  

Also, seems reasonable to remove support for deprecated version `3.4.13` for now, given that the historical commits are still available for time time being. 👍

File: neptune-tinkerpop/.gitignore (hunk 1 above)
Comment (cloutierMat) [reply neptune-tinkerpop/.gitignore:3]: I did, when creating the maven package, a file gets created at `./neptune-tinkerpop/3.6.2/file:/home/matt/.m2/repository/org/apache/tinkerpop/tinkergraph-gremlin/3.6.2/test-case-data/TinkerGraph/resources/tinkerpop-modern-v3d0.kryo`.

**Pull request merged**